net_handler/tcp_conn.c

```diff
@@ -183,6 +184,12 @@ static int recv_new_msg(int conn_fd, int challenge_fd) {
             return -2;
         }
 
+        // if shepherd is connected and dawn tries to send RUN_MODE == AUTO or TELEOP, block it
+        if (pthread_self() == dawn_tid && robot_desc_read(SHEPHERD) == CONNECTED && (run_mode_msg->mode == MODE__AUTO || (run_mode_msg->mode == MODE__TELEOP))) {
+            log_printf(INFO, "You cannot send Robot to Auto or Teleop from Dawn with Shepherd connected!");
```

Comment:
Very minor Q, but did you mean to write "set" instead of "send"?

Reply:
I meant to write "send" but you're right, "set" make more sense.